api/server.py

```diff
@@ -166,4 +179,70 @@ def expression():
         body = request.json
         response = database_lib.expression(body).to_list()
         return jsonify(response)
-        
\ No newline at end of file
+
+content_files = [
+    'content.json',
+    'content.staged.json',
+]
+
+import tempfile
+import shutil
+
+def add_content_route(content_file):
+    endpoint = '/api/' + content_file
+    @app.route(endpoint, methods=['POST', 'GET'], endpoint=endpoint)
+    def content():
+        if request.method == 'GET':
+            with open(content_file) as json_file:
+                response = json.load(json_file)
+            return jsonify(response)
+        elif request.method == 'POST':
+            if not is_whitelisted():
+                return redirect(url_for("login"))
```

Comment:
I suspect this endpoint may be called from code rather than being an actual form target/similar. Given that, it may be more reasonable to just flag authentication failure rather than doing a redirect.

Reply:
Correct, this will not work. Fixing...